### PR TITLE
CDP-1568: Repair backup channel audit support for traefik

### PR DESCRIPTION
**Pull request by osterzel**
Repair backup audit channel messages

### CODE REVIEW

File: middlewares/audittap/streams/httpSinkAsync.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-const undeliveredMessagePrefix = "DS_EventMissed_AuditFailureResponse"
+const undeliveredMessagePrefix = "DS_EventMissed_AuditFailureResponse audit item : "
```
Comment (aedwards9) [inline middlewares/audittap/streams/httpSinkAsync.go]: Looking at the datastream equivalent it looks like it needs a colon between the key and audit item too, i.e.

"DS_EventMissed_AuditFailureResponse : audit item : "

File: middlewares/audittap/streams/httpSinkAsync_test.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func TestLogEventsOnNon200Response(t *testing.T) {
 	_ = w1.Audit(encodedJSONSample)
 
 	time.Sleep(1000 * time.Millisecond)
-	assert.True(t, strings.Contains(buf.String(), `{"level":"warning","message":"DS_EventMissed_AuditFailureResponse"`))
+	assert.True(t, strings.Contains(buf.String(), `{"level":"warning","message":"DS_EventMissed_AuditFailureResponse audit item : [1,2,3]"`))
```
Comment (aedwards9) [inline middlewares/audittap/streams/httpSinkAsync_test.go]: As above, colon needed

File: middlewares/audittap/streams/httpSinkAsync.go (hunk 1 above)
Comment (osterzel) [reply middlewares/audittap/streams/httpSinkAsync.go]: This is effectively a revert of what has been there previously and was originally implemented in traefik, I didn't want to change the code to much, you can look at git commit b69c779066934d6b9d1a943ec3c04ae7fd2b9462 which removed this.

The logstash config takes everything up the string "audit item :" and removes it so having that extra : is not really required

Comment (osterzel) [reply middlewares/audittap/streams/httpSinkAsync.go]: Maybe I should add that in to just ensure consistency across the board.

Comment (osterzel) [reply middlewares/audittap/streams/httpSinkAsync.go]: Updated to be consistent with other codebases

File: middlewares/audittap/streams/httpSinkAsync_test.go (hunk 2 above)
Comment (osterzel) [reply middlewares/audittap/streams/httpSinkAsync_test.go]: Updated

File: middlewares/audittap/streams/httpSinkAsync.go [hunk 3]
```diff
@@ -223,5 +224,5 @@ func (p *httpProducerAsync) publish() {
 }
 
 func handleFailedMessage(encoded atypes.Encoded) {
-	log.Warn(undeliveredMessagePrefix)
+	log.Warn(undeliveredMessagePrefix + string(encoded.Bytes))
```
Comment (aedwards9) [inline middlewares/audittap/streams/httpSinkAsync.go:227]: Not a Go expert but there seems to be some debate on whether it should be encoded.Bytes[:]
https://stackoverflow.com/questions/40632802/how-to-convert-byte-array-to-string-in-go

I've just seen you mention that you've just reverted to the old version so I'm guessing it's right but do we have any tests for this? And/or do we have a way of testing it in the non-prod environments? I'm guessing it's difficult as we don't own the running service...

Comment (osterzel) [reply middlewares/audittap/streams/httpSinkAsync.go:227]: The tests that have been changed in the other file check for creation of these events correctly under various scenarios

We can also test this locally by running the traefik services and pushing audits through it. This has been done.

Additionally we should test the whole process in a lower environment so when they deploy it we'll run some tests through to validate

**Review by aedwards9 — APPROVED** [review]

**Pull request merged**